ir/shared/src/main/scala/org/scalajs/ir/Serializers.scala

```diff
@@ -1428,243 +1471,232 @@ object Serializers {
                   s"Found invalid pre-1.11 JS constructor def at ${methodDef.pos}:\n${methodDef.show}")
           }
 
-        case _ =>
-          memberDef
+        case exportedMember =>
+          exportedMembersBuilder += exportedMember
       }
+
+      (jsConstructorDefBuilder.result(), exportedMembersBuilder.result())
     }
 
-    def readMemberDef(owner: ClassName, ownerKind: ClassKind): MemberDef = {
-      implicit val pos = readPosition()
-      val tag = readByte()
+    private def readFieldDef()(implicit pos: Position): FieldDef = {
```

Comment:
The diff from `readMemberDef` to `readFieldDef+readMethodDef+etc.` is atrocious. :(

Could you perhaps do a first refactoring-only commit that extracts `readFieldDef` and co from the individual `case`s of `readMemberDef`, without touching anything else? The diff for that commit would still be bad, but if it's strictly only moving things in extracted methods, it's easy to review and more importantly easy to ignore in a future `git blame`. Then the second commit will have a nice diff here.